pkg/cloudprovider/azure.go

```diff
@@ -339,29 +350,30 @@ func (a *Azure) GetNodeEgressIPConfiguration(node *corev1.Node, cloudPrivateIPCo
 	return []*NodeEgressIPConfiguration{config}, nil
 }
 
-func (a *Azure) createOrUpdate(networkInterface network.Interface) (network.InterfacesCreateOrUpdateFuture, error) {
+func (a *Azure) createOrUpdate(networkInterface armnetwork.Interface) error {
 	ctx, cancel := context.WithTimeout(a.ctx, defaultAzureOperationTimeout)
 	defer cancel()
-	return a.networkClient.CreateOrUpdate(ctx, a.resourceGroup, *networkInterface.Name, networkInterface)
-}
+	_, err := a.networkClient.BeginCreateOrUpdate(ctx, a.resourceGroup, ptr.Deref(networkInterface.Name, ""), networkInterface, nil)
+	if err != nil {
+		return fmt.Errorf("failed to create or update network interface: %v, err: %v", ptr.Deref(networkInterface.Name, ""), err)
+	}
 
-func (a *Azure) waitForCompletion(result network.InterfacesCreateOrUpdateFuture) error {
-	// No specified timeout for this operation, because a valid value doesn't
-	// seem possible to estimate. Note: Azure has some defaults defined here:
-	// https://github.com/Azure/go-autorest/blob/master/autorest/client.go#L32-L44
-	return result.WaitForCompletionRef(context.TODO(), a.networkClient.Client)
+	return nil
 }
 
-func (a *Azure) getSubnet(networkInterface network.Interface) (*net.IPNet, *net.IPNet, error) {
+func (a *Azure) getSubnet(networkInterface armnetwork.Interface) (*net.IPNet, *net.IPNet, error) {
 	addressPrefixes, err := a.getAddressPrefixes(networkInterface)
 	if err != nil {
 		return nil, nil, fmt.Errorf("error retrieving associated address prefix for network interface, err: %v", err)
 	}
 	var v4Subnet, v6Subnet *net.IPNet
 	for _, addressPrefix := range addressPrefixes {
-		_, subnet, err := net.ParseCIDR(addressPrefix)
+		if addressPrefix == nil {
+			return nil, nil, fmt.Errorf("error retrieving associated address prefix")
+		}
+		_, subnet, err := net.ParseCIDR(ptr.Deref(addressPrefix, ""))
```

Comment:
what is the point of `ptr.Deref` if you checked for nil above?

Reply:
There is not, but I was just cleaning them all up from earlier PR comments. I will remove it since you pointed it out.